application-task-manager-default/src/main/java/com/xwiki/taskmanager/internal/DefaultTaskManager.java

```diff
@@ -0,0 +1,128 @@
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xwiki.taskmanager.TaskException;
+import com.xwiki.taskmanager.TaskManager;
+import com.xwiki.taskmanager.model.Task;
+
+/**
+ * Default implementation of {@link TaskManager}.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Singleton
+public class DefaultTaskManager implements TaskManager
+{
+    private static final LocalDocumentReference TASK_CLASS_REFERENCE = new LocalDocumentReference(Arrays.asList(
+        "TaskManager", "Code"), "TaskClass");
+
+    @Inject
+    private DocumentReferenceResolver<String> resolver;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private QueryManager queryManager;
+
+    @Inject
+    private EntityReferenceSerializer<String> serializer;
+    @Inject
+    private Logger logger;
+
+    @Override
+    public Task getTaskByReference(String reference)
+    {
+        XWikiContext context = contextProvider.get();
+        DocumentReference docRef = resolver.resolve(reference);
+        try {
+            XWikiDocument doc = context.getWiki().getDocument(docRef, context);
+            Task task = new Task();
+            BaseObject obj = doc.getXObject(TASK_CLASS_REFERENCE);
+            if (obj == null) {
+                logger.warn("The page [{}] does not have a Task Object.", docRef);
+                return null;
+            }
+            task.setReference(docRef);
+            task.setName(obj.getStringValue(Task.NAME));
+            task.setNumber(obj.getIntValue(Task.NUMBER));
+            task.setOwner(resolver.resolve(obj.getLargeStringValue(Task.OWNER)));
+            task.setAssignee(resolver.resolve(obj.getLargeStringValue(Task.ASSIGNEE)));
+            task.setStatus(obj.getStringValue(Task.STATUS));
+            task.setReporter(resolver.resolve(obj.getLargeStringValue(Task.REPORTER)));
+            task.setRender(obj.getLargeStringValue(Task.RENDER));
+            task.setDuedate(obj.getDateValue(Task.DUE_DATE));
+            task.setCreateDate(obj.getDateValue(Task.CREATE_DATE));
+            task.setCompleteDate(obj.getDateValue(Task.COMPLETE_DATE));
+            return task;
+        } catch (XWikiException e) {
+            logger.error("Failed to retrieve the task from the page [{}]", reference);
+            return null;
+        }
+    }
+
+    @Override
+    public void deleteTasksByOwner(DocumentReference documentReference) throws TaskException
+    {
+        try {
+            XWikiContext context = contextProvider.get();
+            String statement = ", BaseObject as taskObj, StringProperty as ownerProp "
+                + "WHERE taskObj.name = doc.fullName AND taskObj.className = 'TaskManager.Code.TaskClass' "
+                + "AND taskObj.id = ownerProp.id.id AND ownerProp.id.name = 'owner' "
+                + "AND ownerProp.value like :docRef";
+            Query query = queryManager.createQuery(statement, Query.HQL);
```

Comment:
```suggestion
            String statement = "FROM doc.object(TaskManager.Code.TaskClass) as task WHERE task.owner = :absoluteOwnerRef OR task.owner = :compactOwnerRef OR (task.owner = :relativeOwnerRef AND doc.space = :ownerSpaceRef)";
            Query query = queryManager.createQuery(statement, Query.XWQL);
```

Using XWQL makes the query simpler. Also, I'm not sure about the like operator. Doesn't it match too much? I think it's safer to match against all possible references: absolute (``wiki:Path.To.Owner``), compact (``Path.To.Owner``) and relative (``Owner``). Also, before deleting, I'd double check that the owner of the return task document matches the given owner reference (input parameter), by resolving the owner value relative to the task document reference, just to be sure.